src/audio/pipeline.c

```diff
@@ -532,6 +532,34 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 
 	trace_pipe_with_ids(p, "pipeline_trigger()");
 
+	/* handle pipeline global checks before going into each components */
```

Comment:
This is all fine, except lets move this xrun handling to a different function i..e.
```
pipeline_handle_xrun_trigger()
{
    // return value indicates error, caller can continue or caller should return
}

pipeline_trigger()
{
    if (p->in_xrun) {
        ret = pipeline_handle_xrun_trigger();
        if (ret < 0)
            error;
        else if (ret > 0)
           return ret; // xrun handler handled this no further action needed.
      }
    .....
}
```

Reply:
@lrgirdwo updated as you suggested.